### ISSUE TITLE
Release - August 2018 - Target RC date: 2018-08-01 - name: 0.17.0

### ISSUE THREAD

**Issue by dslomov** [bazelbuild/bazel#5059]

**Comment by dslomov** [comment]
@jin lmk if you are not available on these dates

**Comment by dkelmer** [comment]
Can we delay cutting the release until https://github.com/bazelbuild/bazel/issues/3115 is fixed? I'm aiming to mail a fix today

**Comment by buchgr** [comment]
@dkelmer if it's a regression we can also cherry pick it

**Comment by dkelmer** [comment]
That works too. rules_go is definitely broken and I imagine other projects are too

**Comment by jin** [comment]
The latest nightly looks pretty good: https://buildkite.com/bazel/bazel-with-downstream-projects-bazel/builds/366

Projects in the red:

- [ ] rules_apple
- [ ] bazel_toolchains
- [ ] rules_go
- [ ] Tensorflow
- [ ] rules_python

**Comment by jin** [comment]
```scripts/release/release.sh create 0.17.0 afebc60a337e98aa7b5b4f45cc809dae5cd0c765 510aca0c993d89cccdf2da1b8bcafff35a0d030f```

**Comment by petemounce** [comment]
Pushed to chocolatey.

**Comment by jin** [comment]
RC1 pushed https://releases.bazel.build/0.17.0/rc1/index.html

**Comment by drigz** [comment]
Can I suggest adding a blocker for some simple migration instructions for #4502? As you can see it breaks lots of projects, and there are questions on #4502 and [one on bazel-discuss](https://groups.google.com/forum/?pli=1#!topic/bazel-discuss/IWEA7npKNYg) about how to migrate.

**Comment by cvcal** [comment]
Yes, we'll make sure there are migration instructions in the release notes.
Is there somewhere else you think we should add instructions?

**Comment by buchgr** [comment]
@jin I think due to the regressions in the 0.16.0 release regarding the embedded JDK and Java rules we'll need to put the 0.17.0 release on hold for now.

**Comment by drigz** [comment]
@cvcal I guess there's a chicken/egg problem, but it would be ideal to link them from the warning. Otherwise a noticeable mention in the release notes seems sufficient.

PS. The warning is not colored for me, so it doesn't stand out so much from the stdout-spam[1], is that expected?

![image](https://user-images.githubusercontent.com/632207/43783251-88027862-9a61-11e8-81cc-dae33f044ce1.png)

[1]: I have, to my great shame, failed to stop this building up in our project.

**Comment by cvcal** [comment]
Unfortunately, yes - the warning is printed from the C++ client, not the server, which is where the colored lines come from. I know this makes it less noticeable, and that's sad... 
I could get around this by sending the server `--unconditional_warning=blah` but adding command arguments from the client is something we really try not to do

**Comment by drigz** [comment]
No problem, thanks for the clarification. Maybe other people have cleaner
builds :)

Thanks for being so responsive the whole way through on this change by the
way!

**Comment by jin** [comment]
As of @dslomov's email in https://groups.google.com/forum/#!topic/bazel-discuss/1116GHzkNm0, we'll put 0.17.0 on hold until we cut a new 0.16.1 release with the fixes to JDK 9.

**Comment by buchgr** [comment]
We are expecting to release 0.16.1 on Monday, 13th August based on this release candidate https://releases.bazel.build/0.16.1/rc2/index.html

**Comment by davido** [comment]
> we'll put 0.17.0 on hold until we cut a new 0.16.1 release with the fixes to the embedded JDK 9.

It seems that Gerrit cannot be built any more on JDK9 on 0.17.0rc1 and master, see #5896.

**Comment by jin** [comment]
Reassigning to the new release team @aehlig @philwo @katre @laurentlb

**Comment by laurentlb** [comment]
What is the current state of 0.17? I see that 0.16.1 has been released (https://github.com/bazelbuild/bazel/releases).

I can cut a new 0.17, with the same baseline and cherrypick as https://releases.bazel.build/0.17.0/rc1/index.html. On top of that, I can also cherrypick everything that went into 0.16.1. Is that correct?

**Comment by cvcal** [comment]
Issues #5765 and #5906 have a pending fix which should get included in 0.17

**Comment by laurentlb** [comment]
@cvcal @buchgr 
Please update the thread when the changes are submitted (with commit hashes).

**Comment by jin** [comment]
Considering that there are a ton of cherrypicks to 0.16.1 (https://github.com/bazelbuild/bazel/issues/5792) that happened _after_ cutting 0.17.0rc1, maybe it's more worth to recut 0.17.0 at a new baseline, or abandon 0.17.0 and cut 0.18.0 directly.

**Comment by laurentlb** [comment]
Yes, I'd like to use a new baseline (and call it `0.17.0` since it has not been released).

**Comment by buchgr** [comment]
@laurentlb please cut after f0aa55d81c7a311efef203bdecf7bca99aaded00

**Comment by laurentlb** [comment]
@cvcal Can I cut the new release now?

**Comment by ola-rozenfeld** [comment]
I see https://github.com/bazelbuild/bazel/pull/5898 is still not merged.

**Comment by ola-rozenfeld** [comment]
Actually, can someone please merge https://github.com/bazelbuild/bazel/pull/5917 as a prerequisite to 0.17? This is a fix to a regression in handling actions that were garbage collected from remote cache, and it is affecting Buildfarm. I reviewed it, but I don't have merge permissions. Thank you!

**Comment by cvcal** [comment]
Sorry, getting the warning fix in lead to a few windows-only issues that I wasted time on. #5765 is pretty close to submitted

**Comment by laurentlb** [comment]
We can't delay the release any longer. I'll cut it soon.
(changes can probably wait for the next release)

**Comment by laurentlb** [comment]
Bazel 0.17rc1 with baseline aa118ca818baf722aede0bc48d0a17584fa45b6e
https://releases.bazel.build/0.17.0/rc1/index.html

Review and contribute to the release notes: https://docs.google.com/document/d/1wDvulLlj4NAlPZamdlEVFORks3YXJonCjyuQMUQEmB0/edit#heading=h.fdnt5bpp42hj

**Comment by davido** [comment]
> Bazel 0.17rc1 with baseline aa118ca https://releases.bazel.build/0.17.0/rc1/index.html

I've just tested 0.17rc1 and confirmed that trying to Build Gerrit on JDK9 [is failing in the same way](https://github.com/bazelbuild/bazel/issues/5962) as  Bazel@HEAD.

**Comment by jin** [comment]
@laurentlb using the same 0.17.0rc1 tag for two different releases has some issues here: https://github.com/bazelbuild/bazel/issues/5964

**Comment by laurentlb** [comment]
Created yesterday 0.17.1: https://releases.bazel.build/0.17.1/rc1/index.html

**Comment by jin** [comment]
https://github.com/bazelbuild/bazel/issues/5978 is a release blocker -- 0.17.1 and HEAD binaries that are built with 0.16.1 contains a `--javabase` linking bug.

**Comment by davido** [comment]
I've found this regression in 0.17.0rc1 with error prone integration: #5989, that was fixed in #5990. Can the fix be approved, merged and cherry-picked?

**Comment by davido** [comment]
> 

Another CL was merged that fixed that regression: 0e0462589528154cb5160411991075a2000b5452. Can this be cherry-picked to unblock users who heavily rely on working error-prone integration?

**Comment by katre** [comment]
Release blocker: #6020
The debian package is not correctly setup for conffiles, and is overwriting existing global bazelrc files.
I'll have a fix out shortly, but we should cherrypick it to this release.

**Comment by davido** [comment]
There is another regression in 0.17-rc1: rules_closure (and Gerrit) cannot be built any more with `@bazel_tools//tools/jdk:toolchain_java9`: #5961. It was fixed in: https://github.com/bazelbuild/bazel/commit/3987300d6651cf0e6e91b395696afac6913a7d66 . Can this fix be cherry-picked?

**Comment by katre** [comment]
The commits 2147666fa2750b4999fff763f6e4e3d016d6cc9e and 438b2773b8c019afa46be470b90bcf70ede7f2ef need to be cherrypicked into the next RC (to fix #6020 and #6028 respectively).

**Comment by laurentlb** [comment]
To summarize, here's the current list of cherrypicks:
https://github.com/bazelbuild/bazel/commit/0e0462589528154cb5160411991075a2000b5452
https://github.com/bazelbuild/bazel/commit/3987300d6651cf0e6e91b395696afac6913a7d66
https://github.com/bazelbuild/bazel/commit/438b2773b8c019afa46be470b90bcf70ede7f2ef
https://github.com/bazelbuild/bazel/commit/2147666fa2750b4999fff763f6e4e3d016d6cc9e
https://github.com/bazelbuild/bazel/commit/9ed9d8ac4347408d15c8fce7c9c07e5c8e658b30
(let me know if I forgot one)

**Comment by laszlocsomor** [comment]
Confirmed: https://github.com/bazelbuild/bazel/issues/6061 is a blocker. (Release 0.16.1 shipped with Protobuf 3.4.0, the 0.17.* release candidates with Protobuf 3.6.0 which contains a bug.)

**Comment by laszlocsomor** [comment]
Please cherrypick: https://github.com/bazelbuild/bazel/commit/22d761ab42dfb1b131f1facbf490ccdb6c17b89c
https://github.com/bazelbuild/bazel/commit/27303d79c38f2bfa3b64ee7cd7a6ef03a9a87842
https://github.com/bazelbuild/bazel/commit/ddc97ed6b0367eb443e3e09a28d10e65179616ab

to fix https://github.com/bazelbuild/bazel/issues/6061 (and transitively https://github.com/bazelbuild/bazel/issues/6053).

**Comment by laszlocsomor** [comment]
We need one more cherrypick, because https://github.com/bazelbuild/bazel/commit/27303d79c38f2bfa3b64ee7cd7a6ef03a9a87842 should have been imported but it wasn't. @mhlopko is preparing a fix. Stay tuned.

**Comment by laszlocsomor** [comment]
Please also cherry-pick https://github.com/bazelbuild/bazel/commit/ead1002d3803fdfd4ac68b4b4872076b19d511a2. There's a chance that it will be an empty cherry-pick, because this commit is replicating parts of https://github.com/bazelbuild/bazel/commit/27303d79c38f2bfa3b64ee7cd7a6ef03a9a87842 (the parts not mirrored in Google's internal version control).

**Comment by steeve** [comment]
Is it possible to cherry-pick 4a6891f83efc1b1891bc994d53722225e8acdd77 please ? Thanks!

**Comment by laurentlb** [comment]
We do cherrypicks only for regressions. I think this commit will have to wait until Bazel 0.18 (that shouldn't be too much wait since 0.17 is late).

**Comment by ittaiz** [comment]
When is RC2 planned? We've had several releases which had 6-7 RCs

**Comment by mhlopko** [comment]
Please consider cherrypicking fa2d3f60f5cce16e414aa843b26f379a72b8f7b0. It fixes a bug released in 0.10. More context: https://groups.google.com/forum/#!topic/bazel-discuss/NciKWGqdPzM

**Comment by lberki** [comment]
Please cherrypick these changes:

https://github.com/bazelbuild/bazel/commit/12dcd35ef7a26d690589b0fbefb1f20090cbfe15
https://github.com/bazelbuild/bazel/commit/7eb9ea150fb889a93908d96896db77d5658e5005
https://github.com/bazelbuild/bazel/commit/4566a428c5317d87940aeacfd65f1018340e52b6
https://github.com/bazelbuild/bazel/commit/1e9f0aa89dad38eeab0bd40e95e689be2ab6e5e5

To revert the embedded JDK and java compiler to version 9.

**Comment by laurentlb** [comment]
Release candidate 2 is here: https://releases.bazel.build/0.17.1/rc2/index.html

@ittaiz I was waiting for the JDK fix before creating rc2 (but maybe I should have created it anyway).

**Comment by laszlocsomor** [comment]
0.17.0 rc2 regression on Windows: https://github.com/bazelbuild/bazel/issues/6099

**Comment by philwo** [comment]
Can we please cherry-pick these two updates for the CI configuration?

- a572c1cbc8c26f625cab6716137e2d57d05cfdf3
- 29f1de099e4f6f0f50986aaa4374fc5fb7744ee8

They are *totally* risk free. ;)

**Comment by philwo** [comment]
Also this one for the updated bazel_toolchains repo:

- b495eafdc2ab380afe533514b3bcd7d5b30c9935

Fixes the error "no such package '@bazel_toolchains//configs/ubuntu16_04_clang/1.0/bazel_0.16.1/default': BUILD file not found on package path" on CI.

**Comment by laurentlb** [comment]
rc3 created with:
`scripts/release/release.sh create --force_rc=3 $RELEASE_NUMBER` aa118ca818baf722aede0bc48d0a17584fa45b6e 0e04625 3987300 438b277 504401791e0a0e7e3263940e9e127f74956e7806 9ed9d8a 22d761a 27303d7 ddc97ed ead1002 12dcd35 7eb9ea1 4566a42 1e9f0aa a572c1c 29f1de0 b495eaf

It should behave the same as rc2, the changes are just for testing and CI.

https://releases.bazel.build/0.17.1/rc3/index.html

**Comment by gertvdijk** [comment]
Was it intended to create a 0.17.**1** RC release? I expected a 0.17.**0** RC3. :smile:

**Comment by laurentlb** [comment]
Yes. 0.17.0 was abandoned, 0.17.1 was created using a different baseline.

**Comment by laszlocsomor** [comment]
Please cherrypick https://github.com/bazelbuild/bazel/commit/9323c57607d37f9c949b60e293b573584906da46 to fix https://github.com/bazelbuild/bazel/issues/6099.

**Comment by laurentlb** [comment]
Done, 9323c57607d37f9c949b60e293b573584906da46 added.
New candidate is here: https://releases.bazel.build/0.17.1/rc4/index.html

I think all known issues have been fixed. If nothing new is reported, we should release this week.

**Comment by iirina** [comment]
Please also cherry pick 1aba9ac4b4f68b69f2d91e88cfa8e5dcc7cb98c2. Envoy build breaks without it.

**Comment by laurentlb** [comment]
Done. This is 0.17.1rc5: https://releases.bazel.build/0.17.1/rc5/index.html

Created using:
`scripts/release/release.sh create --force_rc=5 0.17.1` aa118ca818baf722aede0bc48d0a17584fa45b6e 0e04625 3987300 438b277 504401791e0a0e7e3263940e9e127f74956e7806 9ed9d8a 22d761a 27303d7 ddc97ed ead1002 12dcd35 7eb9ea1 4566a42 1e9f0aa a572c1c 29f1de0 b495eaf 9323c57 1aba9ac4b4f68b69f2d91e88cfa8e5dcc7cb98c2

**Comment by ittaiz** [comment]
FYI- I think #6121 will be a release blocker but I just filed it so obviously no one triaged it yet.

**Comment by petemounce** [comment]
RCs 2, 3, 4, 5 all published to chocolatey.

**Comment by davido** [comment]
@laurentlb There is another release blocker: Gerrit cannot be built on Java 9 any more: https://github.com/bazelbuild/bazel/issues/6127. There is no fix available yet.

**Comment by cushon** [comment]
@laurentlb @lberki the fix for #6127 was submitted as 2579b791c023a78a577e8cb827890139d6fb7534

**Comment by laurentlb** [comment]
Should I delay the release again, or can we use this commit in a patch release?

**Comment by laurentlb** [comment]
Cherrypicked both 2579b791c023a78a577e8cb827890139d6fb7534 and 2834613f93f74e988c51cf27eac0e59c79ff3b8f.

https://releases.bazel.build/0.17.1/rc7/index.html
 
Created with:
`scripts/release/release.sh create --force_rc=7 0.17.1` aa118ca818baf722aede0bc48d0a17584fa45b6e 0e04625 3987300 438b277 504401791e0a0e7e3263940e9e127f74956e7806 9ed9d8a 22d761a 27303d7 ddc97ed ead1002 12dcd35 7eb9ea1 4566a42 1e9f0aa a572c1c 29f1de0 b495eaf 9323c57 1aba9ac4b4f68b69f2d91e88cfa8e5dcc7cb98c2 2579b791c023a78a577e8cb827890139d6fb7534 2834613f93f74e988c51cf27eac0e59c79ff3b8f

**Comment by petemounce** [comment]
In the release notes, are the second and third accurate?

> If the same artifact is generated by two distinct but identical actions, and a downstream action has both those actions' outputs in its inputs, the artifact will now appear twice in the downstream action's inputs. If this causes problems in Skylark actions, you can use the uniquify=True argument in Args.add_args.
> If the same artifact is generated by two distinct but identical actions, and a downstream action has both those actions' outputs in its inputs, the artifact will now appear twice in the downstream action's inputs. If this causes problems in Skylark actions, you can use the uniquify=True argument in Args.add_args.

**Comment by laurentlb** [comment]
I think so. The draft release announcement is here: https://docs.google.com/document/d/1wDvulLlj4NAlPZamdlEVFORks3YXJonCjyuQMUQEmB0/edit#heading=h.fdnt5bpp42hj

(we don't update the generated release notes, so they sometimes have issues)

**Comment by laurentlb** [comment]
I plan to release it tomorrow morning.

Sorry for the delays

**Comment by laurentlb** [comment]
Released: https://github.com/bazelbuild/bazel/releases/tag/0.17.1
(cc @petemounce @ilovezfs)

The announcement will be published on the website later today.

**Comment by buchgr** [comment]
@laurentlb ilovezfs is no longer doing homebrew. as mentioned in the release guide you need to update our own homebrew tap: https://github.com/bazelbuild/homebrew-tap

**Comment by petemounce** [comment]
Pushed to chocolatey.

**Comment by steeve** [comment]
@jin it would be nice if it could include the fix for https://github.com/bazelbuild/bazel/issues/6151 as well

**Comment by jin** [comment]
@steeve unfortunately the cherrypick policy doesn't cover that.

**Comment by steeve** [comment]
@jin that's too bad, since this one generates lots and lots of warnings during Android builds and thus creates a lot of noise and I think will make people think this are not working as they should.

**Comment by laurentlb** [comment]
Please use the new bug for discussing cherrypick requests (https://github.com/bazelbuild/bazel/issues/6164).
